knora/dsplib/models/user.py

```diff
@@ -699,17 +700,56 @@ def getAllUsersForProject(con: Connection, proj_shortcode: str) -> list[Any]:
             for proj in project:
                 if proj["id"] == "http://rdfh.ch/projects/" + proj_shortcode:
                     project_users.append(user)
-        return list(map(lambda a: User.fromJsonObj(con, a), project_users))
 
-    def createDefinitionFileObj(self):
-        user = {
+        # find out the 'groups' and 'projects' of every user
+        for user in project_users:
+            user_iri_esc = urllib.parse.quote_plus(user["id"])
+            project_memberships      = con.get(f'/admin/users/iri/{user_iri_esc}/project-memberships')
+            admin_group_memberships  = con.get(f'/admin/users/iri/{user_iri_esc}/project-admin-memberships')
+            normal_group_memberships = con.get(f'/admin/users/iri/{user_iri_esc}/group-memberships')
+
+            for adm_group in admin_group_memberships['projects']:
+                user['projects'].append(f"{adm_group['shortname']}:admin")
+
+            for project in project_memberships['projects']:
+                if f"{project['shortname']}:admin" not in user['projects']:
+                    user['projects'].append(f"{project['shortname']}:member")
+
+            for group in normal_group_memberships['groups']:
+                project_prefix = group['project']['shortname']
+                user['groups'].append(f"{project_prefix}:{group['name']}")
+
+            user['groups'].reverse()
+
+        # convert to User objects
+        res: list[User] = list(map(lambda a: User.fromJsonObj(con, a), project_users))
```

Comment:
```suggestion
        res: list[User] = [User.fromJsonObj(con, a) for a in project_users]
```

I personally don't like lambdas in python, nor its map implementation, for that matter. So I'd suggest using list comprehensions instead.